4 understanding models content

arynhuck opened the pull request:
Closes issue #4

arynhuck committed:
Content

There.

arynhuck committed:
README and TOC and minor sentence change

Adding it to the link lists. Changed one sentence cuz I forgot to with
the content commit. :grimacing:

arynhuck committed:
Id problem fixed

BUT WHAT DO WE DO ABOUT PRIMARY KEYS?!?! Just add them

helmoski commented on content/efcore/getting-started/understanding-models.md:
```diff
@@ -0,0 +1,46 @@
+# Understanding Data Models 
```
Rename file to match title (understanding-data-models.md)

arynhuck replied:
🌮

helmoski commented on content/efcore/getting-started/understanding-models.md:
```diff
@@ -0,0 +1,46 @@
+# Understanding Data Models 
+ 
+In this lesson, we are going to learn what a data model is and how to manually define a data model. 
```
how to manually define a data model for EF Core.

arynhuck replied:
🌮

helmoski commented on content/efcore/getting-started/understanding-models.md:
````diff
@@ -0,0 +1,46 @@
+ 
+```{C#} 
+public class Book { 
+public int Id { get; set;} 
````
Fix tabbing

arynhuck replied:
🌮

helmoski commented on content/efcore/getting-started/understanding-models.md:
````diff
@@ -0,0 +1,46 @@
+```{C#} 
+public class LibraryContext : DbContext 
+{ 
+public DbSet<Book> Books { get; set; } 
````
Fix tabbing

arynhuck replied:
🌮

helmoski commented on content/efcore/getting-started/understanding-models.md:
````diff
@@ -0,0 +1,46 @@
+Let's look at an example. Imagine we are creating a library application. What types of information would be in such a system? Books and readers are two good examples. To include these types of information in our data model, we first define an entity for each of them like so: 
+ 
+```{C#} 
+public class Book { 
````
Put `{` on its own line

helmoski replied:
(not my preference, but it is the standard in C#)

arynhuck replied:
🌮

helmoski commented on content/efcore/getting-started/understanding-models.md:
```diff
@@ -0,0 +1,46 @@
+ 
+Now that we have defined our data model, we can use EF Core to interact with a matching database. It is important that the data model and database schema match because otherwise EF Core won't be able to translate between the two, and it will just throw an error instead.  
+ 
+When mapping the data model to the database, EF Core makes some assumptions which allow it to perform the mapping automatically. For example, it expects each entity in the data model to correspond to a database table that shares the same name as the `DbSet<>` property in the `DbContext`. In our library example, a matching database would have a "Books" table and a "Readers" table. Likewise, it exp ...
```
We should also mention the `Id` properties becoming columns like we do the other properties.

arynhuck replied:
🌮

arynhuck committed:
Rename file. Minor content changes.

Rename to match the title of the tutorial.
Fixed a few sentences and weird tabbing nonsense.

arynhuck commented:
🌮 ?

arynhuck commented:
@helmoski

helmoski commented on content/efcore/getting-started/README.md:
```diff
@@ -15,6 +15,7 @@ This tutorial will cover the following topics:
 
 - [Intro to EF Core](intro-to-ef-core.md)
 - [Installing EF Core](installing-ef-core.md)
+- [Understanding Data Models](understanding-models.md)
```
Update link to match new file name

arynhuck replied:
🌮

helmoski commented on content/efcore/getting-started/toc.md:
```diff
@@ -1,5 +1,6 @@
 - [Intro to EF Core](intro-to-ef-core.md)
 - [Installing EF Core](installing-ef-core.md)
+- [Understanding Data Models](understanding-models.md)
```
Update link to match new file name

arynhuck replied:
🌮

helmoski commented on content/efcore/getting-started/understanding-data-models.md:
````diff
@@ -0,0 +1,48 @@
+```{C#} 
+public class Book 
+{ 
+	public int Id { get; set;} 
````
Use spaces instead of tabs for code examples please

helmoski replied:
If we've already been using tabs everywhere, it might not be worth it

arynhuck replied:
We've been using tabs this whole time... 🙁

helmoski replied:
Ok, that's fine then. It's just annoying how the github diff doesn't display tabs

arynhuck committed:
Renaming things means renaming things

What a fool.
Had to go back and change the references to change the new filename.

arynhuck commented:
:shipit: ?

helmoski commented on content/efcore/getting-started/understanding-data-models.md:
```diff
@@ -0,0 +1,48 @@
+ 
+public class Reader 
+{ 
+	public int Id { get; set;} 
```
Add a space between "set;" and "}"

arynhuck replied:
🌮 
why so nitpicky

helmoski commented on content/efcore/getting-started/understanding-data-models.md:
````diff
@@ -0,0 +1,48 @@
+```{C#} 
+public class Book 
+{ 
+	public int Id { get; set;} 
````
Add a space between "set;" and "}"

arynhuck replied:
🌮

arynhuck committed:
Nitpicky space additions

Spaces.
